posthog/queries/util.py

```diff
@@ -103,7 +103,9 @@ def convert_to_datetime_aware(date_obj):
     return date_obj
 
 
-def correct_result_for_sampling(value: int, sampling_factor: Optional[float], entity_math: Optional[str] = None) -> int:
+def correct_result_for_sampling(
+    value: Union[int, float], sampling_factor: Optional[float], entity_math: Optional[str] = None
+) -> int:
```

Comment:
What's going on with the typing here? Should we retain typing here instead of casting-to-int if dealing with floats?

Reply:
yeah I made a mess here fighting with CI - will retain types